src/index.ts

```diff
@@ -67,6 +68,10 @@ export default class ServerlessS3Cleaner implements Plugin {
     const config = this.loadConfig();
     let bucketsToEmpty = isDeploying ? config.bucketsToCleanOnDeploy : config.buckets;
 
+    if (!isDeploying && config.autoResolve) {
+      bucketsToEmpty.push(await this.provider.getServerlessDeploymentBucketName());
```

Comment:
Does this work properly with an explicit bucket name as well as a generated (default) bucket name?

Reply:
In my mind logic is to delete deployment bucket only on `remove` operation according with the rest of listed buckets

Reply:
@jettary  in Serverless Framework you can either let the framework auto-generate the deployment bucket itself, or you can specify an externally managed bucket (see [here](https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml#deployment-bucket)). Does `getServerlessDeploymentBucketName` resolve properly in that case?